astropy/cosmology/io/yaml.py

```diff
@@ -129,7 +131,8 @@ def from_yaml(yml):
     -------
     `~astropy.cosmology.Cosmology` subclass instance
     """
-    return load(yml)
+    with u.add_enabled_units(cu):
```

Comment:
Maybe best for a different PR, but looking at this, I realize your setup is a bit different from things like `imperial`, which have a `u.imperial.enable()` (which does `u.add_enabled_units` under the hood of course).

Reply:
Sounds like a good followup. I recall looking at ``enable`` before and I can't remember why I haven't added the function yet. Perhaps ``imperial`` being inside ``astropy.units`` while ``cosmology.units`` is not and I wasn't/am not sure if the units from `cosmology.units`` should be in the units namespace and accessible by ``u.[unit]``. But I'll look again. I added a card for this
https://github.com/astropy/astropy/projects/7#card-75383377